### PR TITLE
Clean out the now-unneeded has_chunk, has_array, list_chunk_ids

### PR DESCRIPTION
**Pull request by bmerry**

### CODE REVIEW

File: scripts/mvf_rechunk.py [hunk 1]
```diff
@@ -38,6 +38,13 @@ def __init__(self, arg):
             raise ValueError('Chunk sizes must be positive')
 
 
+def _make_lost(data):
+    if all(stride == 0 for stride in data.strides):
```
Comment (ludwigschwardt) [inline scripts/mvf_rechunk.py]: Are you still using the cunning zero strides?

Comment (bmerry) [reply scripts/mvf_rechunk.py]: Ah crap, I thought I'd reverted things here too. Let me see what went wrong.

Comment (bmerry) [reply scripts/mvf_rechunk.py]: Ok, I've force-pushed the right version now (the only changes are in mvf_rechunk.py). Sorry for the inconvenience.

File: scripts/mvf_rechunk.py [hunk 2]
```diff
@@ -48,19 +63,15 @@ def __init__(self, stream_name, array_name, store, chunk_info):
...
-        shape = block_info[None]['chunk-shape']
-        if self.has_data[block_info[None]['chunk-location']]:
-            return np.zeros(shape, np.uint8)
+        raw_data = store.get_dask_array(full_name, chunks, dtype, errors='none')
+        # raw_data has `None` objects instead of ndarrays for chunks with
+        # missing data. That's not actually valid as a dask array, but we use
+        # it to produce lost flags (similarly to datasources.py).
+        self.lost_flags = da.map_blocks(_make_lost, raw_data, dtype=np.uint8)
+        if array_name != 'flags':
+            self.data = da.map_blocks(_fill_missing, raw_data, fill_value, dtype=raw_data.dtype)
```
Comment (ludwigschwardt) [inline scripts/mvf_rechunk.py]: I don't see a definition for `fill_value`...

Comment (bmerry) [reply scripts/mvf_rechunk.py]: Urgh, I've probably made a mess of it yet again. Let me actually test it in the morning and get back to you.

Comment (bmerry) [reply scripts/mvf_rechunk.py]: Ok, should be fixed now, and this time I actually tested it.

**Pull request merged**